arch/arm/src/imxrt/imxrt_flexio.c

```diff
@@ -0,0 +1,909 @@
+ */
+
+static void imxrt_flexio_clear_shifter_status_flags(
+  struct flexio_dev_s *dev,
+  uint32_t mask)
+{
+  uint32_t base = ((struct imxrt_flexiodev_s *)dev)->base;
+
+  putreg32(mask, base + IMXRT_FLEXIO_SHIFTSTAT_OFFSET);
+}
+
+/* Gets the shifter error flags.
+ *
+ * param dev FlexIO device
+ * return Shifter error flags
+ */
+
+static uint32_t imxrt_flexio_get_shifter_error_flags(
+  struct flexio_dev_s *dev)
+{
+  uint32_t base = ((struct imxrt_flexiodev_s *)dev)->base;
+  uint32_t regval;
+
+  regval = getreg32(base + IMXRT_FLEXIO_SHIFTERR_OFFSET);
+
+  return (regval & FLEXIO_SHIFTERR_SEF_MASK);
+}
+
+/* Clears the shifter error flags.
+ *
+ * param dev FlexIO device
+ * param mask The shifter error mask which can be calculated by
+ *            (1 << shifter index)
+ *
+ * note For clearing multiple shifter error flags, for example, two shifter
+ * error flags, can calculate the mask by using ((1 << shifter index0) |
+ * (1 << shifter index1))
+ */
+
+static void imxrt_flexio_clear_shifter_error_flags(
+  struct flexio_dev_s *dev,
+  uint32_t mask)
+{
+  uint32_t base = ((struct imxrt_flexiodev_s *)dev)->base;
+
+  putreg32(mask, base + IMXRT_FLEXIO_SHIFTERR_OFFSET);
+}
+
+/* Gets the timer status flags.
+ *
+ * param dev FlexIO device
+ * return Timer status flags
+ */
+
+static uint32_t imxrt_flexio_get_timer_status_flags(
+  struct flexio_dev_s *dev)
+{
+  uint32_t base = ((struct imxrt_flexiodev_s *)dev)->base;
+  uint32_t regval;
+
+  regval = getreg32(base + IMXRT_FLEXIO_TIMSTAT_OFFSET);
+
+  return (regval & FLEXIO_TIMSTAT_TSF_MASK);
+}
+
+/* Clears the timer status flags.
+ *
+ * param dev FlexIO device
+ * param mask The timer status mask which can be calculated by
+ *            (1 << timer index)
+ *
+ * note For clearing multiple timer status flags, for example, two timer
+ * status flags, can calculate the mask by using ((1 << timer index0) |
+ * (1 << timer index1))
+ */
+
+static void imxrt_flexio_clear_timer_status_flags(
+  struct flexio_dev_s *dev,
+  uint32_t mask)
+{
+  uint32_t base = ((struct imxrt_flexiodev_s *)dev)->base;
+
+  putreg32(mask, base + IMXRT_FLEXIO_TIMSTAT_OFFSET);
+}
+
+/* Enables/disables the shifter status DMA. The DMA request generates when
+ * the corresponding SSF is set.
+ *
+ * For multiple shifter status DMA enables, for example, calculate
+ * the mask by using ((1 << shifter index0) | (1 << shifter index1))
+ *
+ * param dev FlexIO device
+ * param mask The shifter status mask which can be calculated by
+ *            (1 << shifter index)
+ * param enable True to enable, false to disable.
+ */
+
+static void imxrt_flexio_enable_shifter_status_dma(
+  struct flexio_dev_s *dev,
+  uint32_t mask,
+  bool enable)
+{
+  uint32_t base = ((struct imxrt_flexiodev_s *)dev)->base;
+  uint32_t regval;
+
+  if (enable)
+    {
+      regval = getreg32(base + IMXRT_FLEXIO_SHIFTSDEN_OFFSET);
+      regval |= mask;
+      putreg32(regval, base + IMXRT_FLEXIO_SHIFTSDEN_OFFSET);
```

Comment:
There is a pattern in most all the drivers to use static in lines like https://github.com/lupyuen/nuttx/blob/73448fc82e1914a78fd9cd082f40a7cb282cca1c/arch/arm/src/imxrt/imxrt_enet.c#L363-L371